File: backend/src/main/java/com/foodly/foodlyapi/enums/PaymentStatus.java
package com.foodly.foodlyapi.enums;

import lombok.Getter;

@Getter
public enum PaymentStatus {
    PENDING(0, "Beklemede"),
    APPROVED(1, "Onaylandı"),
    REJECTED(2, "Reddedildi"),
    FAILED(3, "Başarısız");

    private final int index;
    private final String description;

    PaymentStatus(int index, String description) {
        this.index = index;
        this.description = description;
    }
}

File: frontend/src/constants/PaymentStatus.js
export const PaymentStatus = {
  PENDING: {
    value: 'PENDING',
    description: 'Beklemede'
  },
  APPROVED: {
    value: 'APPROVED',
    description: 'Onaylandı'
  },
  REJECTED: {
    value: 'REJECTED',
    description: 'Reddedildi'
  }
};

export const getPaymentStatusDescription = (status) => {
  return PaymentStatus[status]?.description || status;
};
